pkg/reconciler/vertex/controller.go

```diff
@@ -615,6 +615,12 @@ func (r *vertexReconciler) findExistingPods(ctx context.Context, vertex *dfv1.Ve
 	return result, nil
 }
 
+func (r *vertexReconciler) isTerminatingPod(ctx context.Context, pod *corev1.Pod) bool {
+	// A pod is terminating if its deletion timestamp is set (i.e., non-nil)
+	// It is also terminating if its status phase is in "Failed" or "Succeeded"
+	return !pod.DeletionTimestamp.IsZero() || pod.Status.Phase == corev1.PodFailed || pod.Status.Phase == corev1.PodSucceeded
```

Comment:
This is a risky behavior. If the pod comes to failed or succeeded status due to a repeatable error, then it keeps creating new pods. Even though they won't use any cpu/memory, but some disk space is still occupied.

Reply:
@Koalk - just thought about it, I think it makes sense to including those 2 phases - I can not think of any normal reason to cause pods keeping going to terminated status (failed or succeeded).

Reply:
Please also update monovertex controller.